### PR TITLE
hidapi: install hidtest only on macOS

### PR DESCRIPTION
**Pull request by iMichka**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Review by SMillerDev — DISMISSED** [review]

File: Formula/hidapi.rb [hunk 1]
```diff
@@ -13,7 +13,8 @@ class Hidapi < Formula
     sha256 cellar: :any, mojave:        "e9c2bec30d5d1e9e0f9f91c43510071ba17234cd968b33f161c56cbee23a4d8d"
   end
 
-  depends_on "autoconf" => :build
+  # autoconf 2.70 fails with: configure.ac:16: error: AC_CONFIG_MACRO_DIR can only be used once
```
Comment (SMillerDev) [inline Formula/hidapi.rb:16]: This should be reported upstream (but we can merge before then)

Comment (iMichka) [reply Formula/hidapi.rb:16]: See https://github.com/libusb/hidapi/issues/264#issuecomment-830914402

Comment (iMichka) [reply Formula/hidapi.rb:16]: I adjusted the comment.

**Comment by jwlarocque** [comment]
For anyone running into `error: AC_CONFIG_MACRO_DIR can only be used once`, it's been fixed in https://github.com/libusb/hidapi/pull/226 so you can just `brew install --HEAD hidapi`.

**Pull request merged**